lib/spack/spack/environment.py

```diff
@@ -1249,6 +1313,53 @@ def _add_concrete_spec(self, spec, concrete, new=True):
         self.concretized_order.append(h)
         self.specs_by_hash[h] = concrete
 
+    def _spec_needs_overwrite(self, spec):
+        # Overwrite the install if it's a dev build (non-transitive)
+        # and the code has been changed since the last install
+        # or one of the dependencies has been reinstalled since
+        # the last install
```

Comment:
Re-inserting comment about update-in-place here: https://github.com/spack/spack/pull/15256#discussion_r491138919 (right now it looks like a change in a dependency will cause a reinstall of all dependents).

Reply:
Yes. This PR does not implement update-in-place, it's just related work that we may want to continue with after this.